### ISSUE TITLE
Double else makes layer unselectable

### ISSUE THREAD

**Issue by MMR100** [qgis/QGIS#33359]
When using rule based styling on a layer and create:

- 1 normal rule (e.g. height > 10 meters)

- 1 else

- 1 more else

The layer features cannot be selected. OK this is not common to have twice "else" in rule based, but it is still a bug. 

Tested on a new project with QGIS 3.4 and QGIS 3.10.1

**Comment by MMR100** [comment]
Tested on QGIS 3.12. The bug remains the same. 
Using "categorised" instead of "rule-based" creates another bug. 
Instead of "else", the categories are named "all other values". When displaying / hiding one of the two "all other values", it can hide all the symbols.

**Comment by roya0045** [comment]
Yeah rule based is wonky when it comes to topological edits and feature based canvas operation from time to time.

**Comment by Pedro-Murteira** [comment]
Still a valid issue on QGIS 3.22.3.
